cinema/views.py

```diff
@@ -24,21 +27,61 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return [
+                permission()
+                for permission in [IsAdminOrIfAuthenticatedReadOnly]
+            ]
+
+        raise exceptions.NotFound
 
 
 class ActorViewSet(viewsets.ModelViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return [
+                permission()
+                for permission in [IsAdminOrIfAuthenticatedReadOnly]
+            ]
+
+        raise exceptions.NotFound
```

Comment:
This is not needed. You need to change inheritance
You can use mixins: https://www.django-rest-framework.org/api-guide/generic-views/#mixins
And then add permission_classes = (IsAdminOrAuthenticatedOrReadOnly, )
**Fix in all places**